src/AspNet.Security.OAuth.Vkontakte/VkontakteAuthenticationHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.WebUtilities;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Vkontakte
+{
+    internal class VkontakteAuthenticationHandler : OAuthHandler<VkontakteAuthenticationOptions>
+    {
+        public VkontakteAuthenticationHandler(HttpClient httpClient) : base(httpClient)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, OAuthTokenResponse tokens)
+        {
+            string endpoint = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, "access_token", tokens.AccessToken);
+
+            if (Options.Fields.Count > 0)
+            {
+                endpoint = QueryHelpers.AddQueryString(endpoint, "fields", string.Join(",", Options.Fields));
+            }
+
+            HttpResponseMessage response = await Backchannel.GetAsync(endpoint, Context.RequestAborted);
+            response.EnsureSuccessStatusCode();
```

Comment:
Please replace that by https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.Gitter/GitterAuthenticationHandler.cs#L32-L39